### ISSUE TITLE
Most StringBuilder tests fail

### ISSUE THREAD

**Issue by ckarenz** [NETMF/llilum#166]
Run TestRunner on K64F.  Resulting output:

```
DebugLog: Test 'Microsoft.Zelig.Test.StringBuilderTests' running...
DebugLog: Failed: Test_0_AppendTest_0
DebugLog: Failed: Test_3_ReplaceTest_0
DebugLog: Failed: Test_3_ReplaceTest_1
DebugLog: Failed: Test_3_ReplaceTest_2
DebugLog: Failed: Test_3_ReplaceTest_3
DebugLog: Failed: Test_3_ReplaceTest_4
DebugLog: Failed: Test_3_ReplaceTest_5
DebugLog: Failed: Test_3_ReplaceTest_6
DebugLog:       StringBuilder("ybp=}sc").Replace("}sc", "[6ltj") returns "ybp=}sc"
DebugLog: Failed: Test_3_ReplaceTest_7
```

**Comment by lt72** [comment]
Fixed with 8c0fbbcab2520574b8374db702f9bb834841352e

**Comment by lt72** [comment]
Now only one test fails, see https://github.com/NETMF/llilum/issues/170
